test/letters/containers/VeteranBenefitSummaryLetter.unit.spec.jsx

```diff
@@ -51,52 +46,93 @@ describe('<VeteranBenefitSummaryLetter>', () => {
   });
 
   it('should show service info options', () => {
-    const tree = SkinDeep.shallowRender(<VeteranBenefitSummaryLetter store={store} {...defaultProps}/>);
-    expect(tree.dive(['div', 'div', '#militaryService'])).not.to.be.empty;
+    const tree = SkinDeep.shallowRender(<VeteranBenefitSummaryLetter {...defaultProps}/>);
+    expect(tree.subTree('#militaryService')).not.to.be.empty;
+  });
+
+  it('renders error and hides benefit table if options not available', () => {
+    const props = _.set('optionsAvailable', false, defaultProps);
+    const tree = SkinDeep.shallowRender(<VeteranBenefitSummaryLetter {...props}/>);
+    const headerText = tree.dive(['.feature', 'h4']).text();
+    expect(headerText).to.equal('Your VA Benefit Summary letter is currently unavailable');
+    expect(tree.subTree('#benefitInfoTable')).to.be.false;
   });
 
-  it('should update request option when checked', () => {
-    const oldWindow = global.window;
+  it('maps each service entry to its own table row', () => {
+    const navyService = [{
+      branch: 'NAVY',
+      characterOfService: 'UNCHARACTERIZED',
+      enteredDate: '1974-01-01T05:00:00.000+00:00',
+      releasedDate: '1976-10-01T04:00:00.000+00:00'
+    }];
+
+    const doubleService = defaultProps.benefitSummaryOptions.serviceInfo.concat(navyService);
+    const props = _.set('benefitSummaryOptions.serviceInfo', doubleService, defaultProps);
+    const tree = SkinDeep.shallowRender(<VeteranBenefitSummaryLetter {...props}/>);
+    const serviceRows = tree.dive(['#militaryServiceTable', 'tbody']).everySubTree('tr');
+
+    expect(serviceRows.length).to.equal(doubleService.length);
+  });
+
+  it('handles check and uncheck events', () => {
+    const oldWindow = { ...global.window };
```

Comment:
This is a shallow clone, so no guarantees this won't mess things up later. :man_shrugging:

I'm pretty sure `global.window.dataLayer` strictly equals `oldWindow.dataLayer` (so they have the same reference) because `dataLayer` is an object. Consequently, this precaution doesn't really do anything. Honestly, it's probably fine for now, but something to keep in mind for if future tests fail unexpectedly. We can always change it to use `_.cloneDeep()` later as a drop-in replacement.

Reply:
Yeah, rather than spend more effort thinking about this I'm just going to deep-clone it :)
This will probably never matter, but would probably be a nasty bug to track down if it did come up.